Source/WebCore/html/canvas/WebGLObject.h

```diff
@@ -90,7 +135,8 @@ class WebGLObject : public RefCounted<WebGLObject> {
     bool m_deleted { false };
 };
 
-inline PlatformGLObject objectOrZero(WebGLObject* object)
+template<typename T>
+PlatformGLObject objectOrZero(const T& object)
```

Comment:
Now we've gone from one function to a specialised version for every derived type of WebGLObject. If this is inlined, that's probably not such a big deal, but it is necessary?

Reply:
Well, we do invoke RefPtr templates, specialized for 80% of types in WebKit, for 80% of the lines that WebKit has..